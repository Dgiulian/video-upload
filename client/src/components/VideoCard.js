import React from 'react';
import { formatSeconds } from '../utils/date';
import TrashFillIcon from './TrashFillIcon';

export default function VideoCard({ video, onVideoSelect, onDelete }) {
  const screenshot = video.screenshot
    ? `${process.env.REACT_APP_API_URL}/screenshots/${video.screenshot}`
    : 'https://dummyimage.com/420x260';
  const duration = formatSeconds(video.duration);
  return (
    <div className="lg:w-1/4 md:w-1/2 p-4 w-full">
      <button
        className="block relative h-48 rounded overflow-hidden"
        onClick={() => onVideoSelect(video._id)}
        data-testid="select"
      >
        <img
          alt="screenshot"
          className="object-cover object-center w-full h-full block"
          src={screenshot}
          data-testid="screenshot"
        />
      </button>
      <div className="flex justify-between items-center">
        <div className="mt-4">
          <h3 className="text-gray-500 text-xs tracking-widest title-font mb-1">
            {video.video_codec} /{' '}
            {video.audio_codec ? video.audio_codec : 'No audio'}
          </h3>
          <h2 className="text-gray-900 title-font text-lg font-medium">
            {video.title}
          </h2>

          <p className="mt-1" data-testid="duration">
            {duration}
          </p>
        </div>
        <button onClick={() => onDelete(video._id)} data-testid="delete">
          <TrashFillIcon className="text-red-600 pointer" />
        </button>
      </div>
    </div>
  );
}
