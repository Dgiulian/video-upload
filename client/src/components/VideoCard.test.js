import React from 'react';
import { render, cleanup, within } from '@testing-library/react';
import VideoCard from './VideoCard';

afterEach(cleanup);

const videoProp = {
  _id: '123456',
  title: 'video title.mp4',
  screenshot: 'screenshot.png',
  duration: 135.8,
  video_codec: 'h264',
  audio_codec: 'aac',
};

describe('Video Card', () => {
  test('it match snapshot', () => {
    const wrapper = render(<VideoCard video={videoProp} />);
    expect(wrapper).toMatchSnapshot();
  });

  test('it calls onDelete function with the appropiate argument', () => {
    const onDelete = jest.fn();
    const { getByTestId } = render(
      <VideoCard video={videoProp} onDelete={onDelete} />
    );
    const deleteButton = getByTestId('delete');
    deleteButton.click();
    expect(onDelete).toHaveBeenCalledTimes(1);
    expect(onDelete).toHaveBeenCalledWith('123456');
  });

  test('it calls onVideoSelect function with the appropiate argument', () => {
    const onVideoSelect = jest.fn();
    const { getByTestId } = render(
      <VideoCard video={videoProp} onVideoSelect={onVideoSelect} />
    );
    const selectButton = getByTestId('select');
    selectButton.click();
    expect(onVideoSelect).toHaveBeenCalledTimes(1);
    expect(onVideoSelect).toHaveBeenCalledWith('123456');
  });
});
