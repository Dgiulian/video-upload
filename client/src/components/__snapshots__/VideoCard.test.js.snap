// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Video Card it match snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="lg:w-1/4 md:w-1/2 p-4 w-full"
      >
        <button
          class="block relative h-48 rounded overflow-hidden"
          data-testid="select"
        >
          <img
            alt="screenshot"
            class="object-cover object-center w-full h-full block"
            data-testid="screenshot"
            src="http://localhost:3005/screenshots/screenshot.png"
          />
        </button>
        <div
          class="flex justify-between items-center"
        >
          <div
            class="mt-4"
          >
            <h3
              class="text-gray-500 text-xs tracking-widest title-font mb-1"
            >
              h264
               /
               
              aac
            </h3>
            <h2
              class="text-gray-900 title-font text-lg font-medium"
            >
              video title.mp4
            </h2>
            <p
              class="mt-1"
              data-testid="duration"
            >
              02:15
            </p>
          </div>
          <button
            data-testid="delete"
          >
            <svg
              class="text-red-600 pointer"
              fill="currentColor"
              height="1em"
              viewBox="0 0 16 16"
              width="1em"
            >
              <path
                d="M2.5 1a1 1 0 00-1 1v1a1 1 0 001 1H3v9a2 2 0 002 2h6a2 2 0 002-2V4h.5a1 1 0 001-1V2a1 1 0 00-1-1H10a1 1 0 00-1-1H7a1 1 0 00-1 1H2.5zm3 4a.5.5 0 01.5.5v7a.5.5 0 01-1 0v-7a.5.5 0 01.5-.5zM8 5a.5.5 0 01.5.5v7a.5.5 0 01-1 0v-7A.5.5 0 018 5zm3 .5a.5.5 0 00-1 0v7a.5.5 0 001 0v-7z"
                fill-rule="evenodd"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="lg:w-1/4 md:w-1/2 p-4 w-full"
    >
      <button
        class="block relative h-48 rounded overflow-hidden"
        data-testid="select"
      >
        <img
          alt="screenshot"
          class="object-cover object-center w-full h-full block"
          data-testid="screenshot"
          src="http://localhost:3005/screenshots/screenshot.png"
        />
      </button>
      <div
        class="flex justify-between items-center"
      >
        <div
          class="mt-4"
        >
          <h3
            class="text-gray-500 text-xs tracking-widest title-font mb-1"
          >
            h264
             /
             
            aac
          </h3>
          <h2
            class="text-gray-900 title-font text-lg font-medium"
          >
            video title.mp4
          </h2>
          <p
            class="mt-1"
            data-testid="duration"
          >
            02:15
          </p>
        </div>
        <button
          data-testid="delete"
        >
          <svg
            class="text-red-600 pointer"
            fill="currentColor"
            height="1em"
            viewBox="0 0 16 16"
            width="1em"
          >
            <path
              d="M2.5 1a1 1 0 00-1 1v1a1 1 0 001 1H3v9a2 2 0 002 2h6a2 2 0 002-2V4h.5a1 1 0 001-1V2a1 1 0 00-1-1H10a1 1 0 00-1-1H7a1 1 0 00-1 1H2.5zm3 4a.5.5 0 01.5.5v7a.5.5 0 01-1 0v-7a.5.5 0 01.5-.5zM8 5a.5.5 0 01.5.5v7a.5.5 0 01-1 0v-7A.5.5 0 018 5zm3 .5a.5.5 0 00-1 0v7a.5.5 0 001 0v-7z"
              fill-rule="evenodd"
            />
          </svg>
        </button>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
